install/powershellscript/README.md

```diff
@@ -0,0 +1,112 @@
+
+On Windows, either Windows PowerShell or PowerShell can be used.
+
+> [!NOTE]
+> PowerShell was formerly known as PowerShell Core.
+> Now [Windows PowerShell and PowerShell](https://learn.microsoft.com/en-us/powershell/scripting/what-is-windows-powershell) are the names used by Microsoft for these products.
+
+## Usage
+
+Run the script using the `pwsh` command:
+
+```bash
+pwsh ./microsoft-go-install.ps1
+```
+
+If you use a PowerShell terminal, you can also choose to run the script directly:
+
+```
+.\microsoft-go-install.ps1
+```
+
+Running the script directly allows the script to change the terminal's `PATH` so the installed Go binary is then available in the current session as `go`.
+
+Note that in typical CI/CD pipelines, each step is run in a fresh process and the `PATH` change will not be preserved.
+If you're using Azure Pipelines, see the help message for `-AzurePipelinePath`.
+
+Pass `-h` to show help.
+
+## Where to put the script
+
+The script can be placed in the root of your repository or in a subdirectory.
+It can be run from a different directory with no effect on the results.
+
+To copy the script and keep it up to date in a reproducible way, use the `microsoftgoinstallscript` command.
+
+### Set up `microsoftgoinstallscript`
+
+In your Go module, run:
+
+```
+go get github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript@latest
+go install github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript
+```
+
+If you don't already use `github.com/microsoft/go-infra` in your module, add a `tools/tools.go` file to your module to pin the version of `microsoftgoinstallscript` and prevent `go mod tidy` from removing it:
+
+```go
+//go:build tools
+
+package tools
+
+import (
+    _ "github.com/microsoft/go-infra/install/powershellscript"
+)
+```
+
+> [!NOTE]
+> This is a well-known workaround to pin the version of a tool in a Go module.
+> See the [Go wiki](https://go.dev/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) for more information.
+
+Then, in the directory of your choice inside your module, run:
+
+```
+microsoftgoinstallscript
+```
+
+### Updating the script using `microsoftgoinstallscript`
+
+To update the script, run this in the directory where the script is stored:
+
+```
+go get github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript@latest
+go mod tidy
+go install github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript
+microsoftgoinstallscript
+```
+
+### Set up a CI test to ensure the script is up to date
+
+First, make sure dependabot is working.
+It will submit PRs that update the microsoft/go-infra dependency automatically.
+
+Then, add a CI test that runs the following commands in the directory where the script is stored:
+
+```
+go install github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript
```

Comment:
Can we omit this command? We already explained how to install `microsoftgoinstallscript` in a previous section, and it should be necessary to reinstall it every time a command is executed.